src/backends/plugins_compat/input_plugin_compat.c

```diff
@@ -78,10 +55,31 @@ uint32_t input_plugin_get_input(void* opaque)
         input.getKeys(control_id, &keys);
     }
 
+    /* disconnect current pak immediately after "pak switch" button is released */
+    if (((keys.Value & PAK_SWITCH_BUTTON) == 0) && ((last_values[control_id] & PAK_SWITCH_BUTTON))) {
+        change_pak(&g_dev.controllers[control_id], NULL, NULL);
+        switch_delay[control_id] = SDL_GetTicks();
```

Comment:
I don't think it's a good idea to use SDL_GetTicks() inside the emulation code. It can lead to varying experiences. What if the user switches paks while fast forward is enabled for instance? It would be better to use count_per_op, similar to how the interrupts are scheduled. This will lead to the same experience no matter how fast/slow the game might be running on the user's machine.

Reply:
I just used the same technique that the SDL input plugin. But you're right, we should avoid that. Maybe a simple counter would be enough.